src/audio/aria/aria_hifi5.c

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation. All rights reserved.
+
+#include "aria.h"
+
+#ifdef ARIA_HIFI5
+#include <xtensa/config/defs.h>
+#include <xtensa/tie/xt_hifi5.h>
+
+/**
+ * \brief Aria gain index mapping table
+ */
+const uint8_t INDEX_TAB[] = {
```

Comment:
this is being cast to a uint32_t in the code and should simplify the L32 loads

Reply:
from the xtensa simulator I didn't see any improvement, the purpose I use the uint8_t is to save memory. But it is okay to change it to a uint32_t table if we don't mind the memory usage.

Reply:
interesting, the L32 of a uint32_t is 1 op, but to load a byte is a L32 and a shift/mask.

Reply:
@lgirdwood @andrula-song , there is load byte instruction, so no need use load32 for this table, save memory is always good:

Load 8-bit Unsigned L8UI
Xtensa Instruction Set Architecture (ISA) Reference Manual 449
Instruction Word (RRI8)
Required Configuration Option
Core Architecture (See Section 4.2 on page 51)
Assembler Syntax
L8UI at, as, 0..255
Description
L8UI is an 8-bit unsigned load from memory. It forms a virtual address by adding the
contents of address register as and an 8-bit zero-extended constant value encoded in
the instruction word. Therefore, the offset ranges from 0 to 255. Eight bits (one byte) are
read from the physical address. This data is then zero-extended and written to address
register at.
If the Region Translation Option (page 185) or the MMU Option (page 199)is enabled,
the virtual address is translated to the physical address. If not, the physical address is
identical to the virtual address. If the translation or memory reference encounters an
error (for example, protection violation or non-existent memory), the processor raises
one of several exceptions (see Section 4.4.2.5 on page 105)